jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -0,0 +1,339 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2012-2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+/**
+ * Bakery holds the context for making HTTP requests
+ * that automatically acquire and discharge macaroons.
+ *
+ * @module env
+ * @submodule env.bakery
+ */
+
+YUI.add('juju-env-bakery', function(Y) {
+
+  var module = Y.namespace('juju.environments.web');
+  var macaroon = Y.namespace('macaroon');
+
+  /**
+   * Bakery client inspired by the equivalent GO code.
+   *
+   * This object exposes the ability to perform requests
+   * that automatically acquire and discharge macaroons
+   *
+   * @class Bakery
+   */
+
+  var Bakery = Y.Base.create('Bakery',
+      Y.Base, [], {
+
+      /**
+       Initialize.
+
+       @method initializer
+       @return {undefined} Nothing.
+       */
+      initializer: function () {
+        this.webhandler = new Y.juju.environments.web.WebHandler();
```

Comment:
We had discussed that you were going to modify this so that it supported having a WebHandler instance passed in as well as creating one internally if it wasn't.